### PR TITLE
Add-logic //Feature events branch

### PR DESCRIPTION
**Pull request by Anvi98**
- Copy the files calculator.js and operate.js into a logic/ directory in your project.
- Analyze the files calculator.js and operate.js, be sure you understand what they do.
- Import the files in your Calculator component.
- Implement the event handlers you need to use the math logic from calculator.js and operate.js in your React component.
- Practice what you have learned until now: use state and props, pass props to child components, use lifecycle methods, lift state up.
- Test your app in the browser. It should be able to perform math operations.

### CODE REVIEW

File: src/components/Calculator.js [hunk 1]
```diff
@@ -1,53 +1,74 @@
 /* eslint-disable no-unused-vars */
+/* eslint-disable max-len */
 import React from 'react';
 import Buttons from './Buttons.js';
 import Input from './Input.js';
+import calculate from '../logic/calculate.js';
 
 class Calculator extends React.Component {
```
Comment (arslanbisharat) [inline src/components/Calculator.js]: - [x]  Please import operate.js here just like calculate.js for the app to work fine.

File: src/components/Calculator.js [hunk 2]
```diff
@@ -1,53 +1,74 @@
...
+    if (next !== undefined) {
+      this.setState({ next });
+    }
+
+    if (total !== undefined) {
+      this.setState({ total });
+    }
+
+    if (operation !== undefined) {
+      this.setState({ operation });
```
Comment (arslanbisharat) [inline src/components/Calculator.js:33]: -  [x] I cannot see division in your project. You have duplicated addition rule. Please work on it.

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
# Changes Requested, 

Hi there, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have limited number of code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/components/Calculator.js (hunk 1 above)
Comment (Anvi98) [reply src/components/Calculator.js]: But Calculator is not using operate.js only calculate.js is using it. So we shouldn't import it here. But anyway if you want.

File: src/components/Calculator.js (hunk 2 above)
Comment (Anvi98) [reply src/components/Calculator.js:33]: Fixed

**Review by arslanbisharat — APPROVED** [review]
# Approved 👏 

Hi there, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of code reviews per project which you can see from your dashbaord. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)_

**Pull request merged**